internal/flushcommon/writebuffer/manager.go

```diff
@@ -96,9 +96,16 @@ func (m *bufferManager) memoryCheck() {
 	if !paramtable.Get().DataNodeCfg.MemoryForceSyncEnable.GetAsBool() {
 		return
 	}
+	startTime := time.Now()
+	m.mut.RLock()
```

Comment:
bettter to:
m.mut.RLock()
defer m.mut.RUnlock()

defer func() { // the warning stuff
}